app/ix/ginas/models/converters/HTMLStringConverter.java

```diff
@@ -88,27 +95,22 @@ public String toStd(String str){
      */
     @Override
     public String truncate(String str, int maxBytes){
-        byte[] b = (str + "   ").getBytes();
-        if(maxBytes >= b.length){
+        if(maxBytes >= str.length()){
             return str;
         }
-        boolean lastComplete = false;
-        int sTag = 0;
+        String truncatedStr = "";
+        StringWriter writer = new StringWriter();
+        Tidy tidy = TIDY_POOL.get();
         for(int i = maxBytes; i >= 0; i--){
-            if(lastComplete || (b[i] & 0x80) == 0) {
-                str = new String(Arrays.copyOf(b, i));
-                sTag = StringUtils.countMatches(str, "<");
-                if(sTag == StringUtils.countMatches(str, ">")
-                    && sTag / 2 == StringUtils.countMatches(str, "/")
-                    && StringUtils.countMatches(str, "&") == StringUtils.countMatches(str, ";")) {
-                    return str;
-                }else{
-                    lastComplete = false;
-                }
+            if (i > 0 && badLastChars.contains(str.charAt(i-1))) {
```

Comment:
This might be slower than the old way and looks like the old way was trying to count close tags?  I assume this means we don't need that anymore

Reply:
check of close tags done by Tidy.